File: gui/wtk/js/ToolkitEnvPeer.js
//
// Copyright (c) 2011, chunquedong
// Licensed under the Academic Free License version 3.0
//
// History:
//   2011-7-4  Jed Young  Creation
//

fan.fgfxWtk.ToolkitEnvPeer = fan.sys.Obj.$extend(fan.sys.Obj);
fan.fgfxWtk.ToolkitEnvPeer.prototype.$ctor = function(self) {}

fan.fgfxWtk.ToolkitEnvPeer.init = function() {
   fan.concurrent.Actor.locals().set("fgfx2d.env", new fan.fgfxWtk.GfxEnv());
   fan.concurrent.Actor.locals().set("fgfxWtk.env", new fan.fgfxWtk.Toolkit());
}

fan.fgfxWtk.Toolkit.prototype.build = function() {
   var win = new fan.fgfxWtk.JsWindow();
   return win;
}

fan.fgfxWtk.Toolkit.prototype.callLater = function(delay, callback)
{
  window.setTimeout(function(){ callback.call(); }, delay);
}

File: gui/wtk/java/fan/fgfxWtk/ToolkitEnvPeer.java
//
// Copyright (c) 2011, chunquedong
// Licensed under the Academic Free License version 3.0
//
// History:
//   2011-7-4  Jed Young  Creation
//

package fan.fgfxWtk;

import javax.swing.*;
import java.awt.*;
import fan.concurrent.Actor;
import fan.sys.*;
import java.util.Timer;
import java.util.TimerTask;


public class ToolkitEnvPeer
{
  public static ToolkitEnvPeer make(ToolkitEnv self)
  {
    ToolkitEnvPeer peer = new ToolkitEnvPeer();
    return peer;
  }

  public static void init()
  {
    Actor.locals().set("fgfx2d.env", AwtGfxEnv.instance);
    Actor.locals().set("fgfxWtk.env", new AwtToolkit());
  }

  static class AwtToolkit extends Toolkit
  {
    Timer timer = new Timer(true);
    public Window build()
    {
      return new AwtWindow();
    }

    public void callLater(final long delay, final Func f)
    {
      TimerTask task = new TimerTask()
      {
        public void run()
        {
          EventQueue.invokeLater(new Runnable()
          {
            public void run()
            {
              f.call();
            }
          });
        }
      };

      timer.schedule(task, delay);
    }
  }
}